### PR TITLE
kubelet/dockertools: Move RunContainer into container runtime.

### PR DESCRIPTION
**Pull request by yifan-gu**
@dchen1107 @vmarmol @bakins

### CODE REVIEW

**Comment by yifan-gu** [comment]
Also removed ContainerRunner interface. 
/cc @pmorie

File: pkg/kubelet/dockertools/manager.go [hunk 1]
```diff
@@ -911,3 +902,35 @@ func (dm *DockerManager) KillContainer(containerID types.UID) error {
 	}
 	return err
 }
+
+// Run a single container from a pod. Returns the docker container ID
+func (dm *DockerManager) RunContainer(pod *api.Pod, container *api.Container, generator kubecontainer.RunContainerOptionsGenerator, runner kubecontainer.HandlerRunner, netMode, ipcMode string) (DockerID, error) {
```
Comment (dchen1107) [inline pkg/kubelet/dockertools/manager.go]: I found myself really start to hate name HandlerRunner :-) Of course no need to change here. Also I don't have a good candidate to replace it.

**Comment by dchen1107** [comment]
LGTM. Will merge on green.

File: pkg/kubelet/dockertools/manager.go (hunk 1 above)
Comment (yifan-gu) [inline pkg/kubelet/dockertools/manager.go]: Sorry :( I am bad at naming..

Comment (pmorie) [inline pkg/kubelet/dockertools/manager.go]: Naming is hard.

**Comment by bakins** [comment]
LGTM

**Comment by vmarmol** [comment]
LGTM

**Pull request merged**